client/tx_config.go

```diff
@@ -34,6 +36,7 @@ type (
 	// signatures, and provide canonical bytes to sign over. The transaction must
 	// also know how to encode itself.
 	TxBuilder interface {
+		gogogrpc.ClientConn
```

Comment:
I don't think an update to ADR 020 is really needed, wdyt?

Reply:
Probably not. But just wondering is it better to change the interface, or create a separate struct that implements ClientConn wraps TxBuilder?

Reply:
I like  the idea of creating ClientConn. I was confused with the `Invoke` method initially.  Maybe we can call the new interface: `MsgSrvClient`?

Reply:
Ping?

Reply:
I'm not sure here. Are you proposing:

```go
type MsgSrvClient struct {
  txBuilder TxBuilder
}

var _ gogogrpc.ClientConn = MsgSrvClient{}
```

? I feel yet another struct created more confusion. At some point we should revisit https://github.com/cosmos/cosmos-sdk/pull/7630#issuecomment-718315664, which imo is the ideal way forward.